### PR TITLE
Added sphinx support

### PR DESCRIPTION
**Pull request by squidarth**
## Added sphinx support for Jupyter Qt Console

* Ran `sphinx-quickstart` to generate a `Makefile` for building documentation.  Docs can now be built by running `make html` in the `docs/` directory.
* Changed the `qtconsole.rst` file to be the index file for this repo
* Added some missing images from the [ipython](https://github.com/ipython/ipython) repo

### CODE REVIEW

**Comment by takluyver** [comment]
There are a couple of files added that I think should be moves - can you remove the old copies?

**Comment by takluyver** [comment]
Test failures are because we need a setup.py - I think @squidarth is working on that for a separate PR.

**Comment by squidarth** [comment]
Are you referring to these?:
 
docs/source/figs/colors_dark.png
docs/source/figs/jn.html
docs/source/figs/jn.xhtml

They were missing from these repo

**Comment by takluyver** [comment]
Yep, my mistake. I think this looks fine now, but I'll let @minrk take a look.

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+# Latex figure (float) alignment
+#'figure_align': 'htbp',
+}
+
+# Grouping the document tree into LaTeX files. List of tuples
+# (source start file, target name, title,
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+  (master_doc, 'JupyterQtConsole.tex', u'Jupyter Qt Console Documentation',
+   u'Jupyter Develompent Team', 'manual'),
```
Comment (minrk) [inline docs/source/conf.py]: s/Develompent/Development/

File: setup.py [hunk 2]
```diff
@@ -88,6 +88,7 @@
 extras_require = setuptools_args['extras_require'] = {
     'test:python_version=="2.7"': ['mock'],
     'test:sys_platform != "win32"': ['pexpect'],
+    'doc': 'Sphinx>=1.1'
```
Comment (minrk) [inline setup.py]: I believe this needs to be a list, not a single string. Is that right?

Comment (minrk) [inline setup.py]: Nope, I'm wrong.

**Comment by minrk** [comment]
One typo, but otherwise AOK.

**Comment by squidarth** [comment]
Nice catch--done

**Comment by takluyver** [comment]
Thanks @squidarth !

**Pull request merged**